src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -191,7 +191,7 @@ public async Task<List<string>> DockerPS(IExecutionContext context, string optio
 
         public async Task<int> DockerNetworkCreate(IExecutionContext context, string network)
         {
-            if (context.StepTarget().ImageOS == PlatformUtil.OS.Windows)
+            if (context.StepTarget()?.ImageOS == PlatformUtil.OS.Windows)
```

Comment:
If StepTarget can return null, should we be falling back to the OS of the agent here?

Reply:
This line of code might need some more thought. From what I can gather, Linux containers won't work with `--driver nat` when running on a Windows host. So we would potentially need to consider all containers that could run in any job (since only one network is created for all steps). 

I might need to defer to @jtpetty on this one since he was the one who originally added this.